packages/calcite-components/src/components/tile-group/tile-group.tsx

```diff
@@ -151,7 +151,20 @@ export class TileGroup implements InteractiveComponent, SelectableGroupComponent
   };
 
   private updateSelectedItems = (): void => {
-    this.selectedItems = this.items?.filter((el) => el.selected);
+    const selectedItems = this.items?.filter((el) => el.selected);
```

Comment:
Yes, `this.items` can be `undefined` since its populated by the slotted tiles and that can't always be guaranteed.  `getSlottedTiles` doesn't do anything extra to prevent from returning `undefined` in some cases, but we could add that in I suppose.

Reply:
I think it's worth getting rid of those guards by updating `getSlottedTiles` to always return an array.